File: 1365. How Many Numbers Are Smaller Than the Current Number/solution.py
class Solution(object):
  def smallerNumbersThanCurrent(self, nums):
    """
    :type nums: List[int]
    :rtype: List[int]
    """
    dictionary = {}
    sortedNums = sorted(nums)
    for index in range(len(sortedNums) - 1, -1, -1):
      num = sortedNums[index]
      dictionary[num] = index
    
    result = nums
    for index, num in enumerate(result):
      result[index] = dictionary[num]

    return result

File: 1365. How Many Numbers Are Smaller Than the Current Number/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
  let dictionary = {};
  let sortedNums = [...nums]
  sortedNums.sort(function(a, b){return a-b});
  for (let index = sortedNums.length - 1; index > -1; index--){
    let num = sortedNums[index];
    dictionary[num] = index;
  }
  
  let result = nums;
  for (let index = 0; index < result.length; index++){
    let num = result[index];
    result[index] = dictionary[num];
  }

  return result;
};